src/Symfony/Bridge/Twig/Extension/WorkflowExtension.php

```diff
@@ -45,6 +48,52 @@ public function getEnabledTransitions($object, $name = null)
         return $this->workflowRegistry->get($object, $name)->getEnabledTransitions($object);
     }
 
+    public function isActive($object, $state, $name = null)
+    {
+        $workflow = $this->workflowRegistry->get($object, $name);
+        $marking = $workflow->getMarking($object);
+
+        return array_key_exists($state, $marking->getPlaces());
+    }
+
+    public function isComplete($object, $state, $name = null)
```

Comment:
You cannot determine if a state is completed just because of the active states, this function tracks back and checks.

Use case: http://bootsnipp.com/snippets/featured/form-process-steps